### ISSUE TITLE
Allow hyper to start on system startup in the background.

### ISSUE THREAD

**Issue by SirWindfield** [zeit/hyper#2775]
I find myself often opening and closing terminal windows as I need them. Therefore a fast startup time would be good to have. Hyper still needs around 2-3 seconds depending on what I am currenty doing. Compared that to the normal OS X Terminal which is available under 1 sec, that is "a lot" of difference.

I thought about an option in the config to make Hyper start on startup and be kept in the background, having the `show` flag on the `BrowserWindow` instance set to false. Once a User tries to start it, the window will be created. I am not sure if this will cut out a lot of time but I can imagine that having Electron already loaded up and just turning the window to `show: true` should be significantly faster. 

It will eat resources while being in the background for sure but that is why I would put this behaviour behind a boolean settings flag.

Any feedback on this? I am not actually sure if this is a rather good idea or just introduces unnecessary complexity. 

Have a nice day ~

**Comment by yogeshkotadiya** [comment]
+1
I would love to have this feature as I also open/close terminal frequently.
About memory, Hyper uses around ~120 MB RAM on my machine (Windows 10) which is really high compared to the native terminal that uses just about 5Mb, not surprised tho knowing that its an electron app.

**Comment by SirWindfield** [comment]
@yogeshkotadiya I do not really mind about the RAM usage nowadays. Most people have probably anyway 8GB+. I might try to see how much impact this has on performance once I have some more time.

**Comment by albinekb** [comment]
If you're concerned about 120 MB ram you should not be using Hyper, our main focus is not to make a resource focused terminal, there's other projects that do that 👍

**Comment by yogeshkotadiya** [comment]
@SirWindfield Agreed. I don't mind 120Mb memory in the background as I have Chrome running in the background which takes around 1Gig memory, I have a 16Gig machine so don't care that much.

@albinekb I don't hate Hyper, in fact I'm using Hyper as my main terminal for the last couple of months and i really like using it. But if you just add the feature to continue running Hyper in background it can load quickly as I tend to quit terminal rather than minimizing it.

I have no idea why even I compared Hyper to a native terminal. That was just stupid 😅.
